assets/thanos-ruler/thanos-ruler.yaml

```diff
@@ -5,6 +5,9 @@ metadata:
     thanosRulerName: user-workload
   name: user-workload
   namespace: openshift-user-workload-monitoring
+  selectorLabels:
```

Comment:
hmm not sure why this gets added :)

Reply:
I had to add that because of the comment https://github.com/openshift/cluster-monitoring-operator/pull/1478#discussion_r751215716

If we don't want to update the selector on the service, then we need to set this in order to get the labels on the service to match the pods in the stateful set.

Basically I don't think we can pick up the defaults here and retain our existing service selector without failing our e2e tests. [Sample](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/1478/pull-ci-openshift-cluster-monitoring-operator-master-e2e-agnostic-operator/1460246190468632576)